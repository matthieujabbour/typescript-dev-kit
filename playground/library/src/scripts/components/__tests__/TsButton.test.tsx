/**
 * @jest-environment jsdom
 */

import React from 'react';
import { act } from 'react-dom/test-utils';
import TsButton from 'scripts/components/TsButton';
import { render, unmountComponentAtNode } from 'react-dom';

let container = document.createElement('div');

describe('react/TsButton', () => {
  beforeEach(() => {
    container = document.createElement('div');
    document.body.appendChild(container);
    jest.clearAllMocks();
  });

  afterEach(() => {
    unmountComponentAtNode(container);
    container.remove();
    ((container as unknown) as null) = null;
  });

  test('renders correctly - basic', () => {
    act(() => {
      render(<TsButton label="Test" />, container);
    });
    expect(container).toMatchSnapshot();
  });
});
