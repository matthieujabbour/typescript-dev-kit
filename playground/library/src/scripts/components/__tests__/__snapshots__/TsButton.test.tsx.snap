// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`react/TsButton renders correctly - basic 1`] = `
<div>
  <button
    type="button"
  >
    <p
      class="ui-p"
    >
      test
    </p>
    Test
  </button>
</div>
`;
